### DISCUSSION TITLE
Integrate sync with external JSON (read-only)

### DISCUSSION THREAD

**Discussion by senthil-kk20** [six7/figma-tokens#164] [Ideas]
Hi Jan,

Just an idea, can we have options to configure an API call from our plugin. So that I can have an API using Node which constantly returns updated JSON which can be updated by anyone online. The idea is to have a page with forms to fill in token data which can be updated by both designer and developer and publish the form to generate JSON. The same JSON will be connected with our plugin and also pushed to Git. Less headache from our side to handle the repository conflicts. Our plugin can show an Update button enabled if there is any update detected so that the designer can agree to update. Your thoughts? 

Thanks,
Senthil

**Comment by six7** [comment]
I like the idea and have heard that feature request before, I think it's a good way to have that common source of truth.

I would keep it read-only to keep it simple, but I guess that's what you meant (correct my if I'm wrong).

Would you expect to be able to provide credentials? Or would that external JSON be public?

That feature could exist in multiple phases:
1) Read only, only public
2) Read only, add credentials
3) Allow Write

**Reply by senthil-kk20** [reply]
We would like to keep it private and not be public. So, it will be "Read only and credentials" since the editing can happen on the site.

**Reply by six7** [reply]
Got it! I'll work on more sync options once the next major release is out, which should include external JSON and git 🙏
